src/Database/ODBC/Internal.hs

```diff
@@ -773,6 +776,43 @@ getData dbc stmt i col = fmap (col, ) $
                    (fmap fromIntegral (odbc_TIME_STRUCT_hour datePtr)) <*>
                    (fmap fromIntegral (odbc_TIME_STRUCT_minute datePtr)) <*>
                    (fmap fromIntegral (odbc_TIME_STRUCT_second datePtr))))
+     | colType == sql_ss_timestampoffset ->
+       withCallocBytes
+         20 -- The TIMESTAMPOFFSET_STRUCT contains 3 SQLSMALLINTs,
```

Comment:
Minor nit pick: I would use the `#size` (from hsc2hs) macro to compute it ?

Reply:
Agreed that the size constant is not ideal.  Since you're OK with it I will leave it as is, rather than introduce hsc2hs.